fix(npm): use parsed body in case response.body is gzip encoded

PSanetra opened the pull request:
Hi,

_sometimes_ the npm login server encodes its response as gzip. If you used username + password + otp, this error was visible as _"Invalid authentication code"_. If you used only username + password as credentials, this probably resulted in the message _"Could not login to npm"_ (did not test this case). 

logged response with gzip encoding:
```
{ statusCode: 201,
  body: <Buffer ... >,
  headers: 
   { 'content-type': 'application/json',
     'content-encoding': 'gzip',
     'cache-control': 'max-age=300'
     [...]
   },
   [...]
}
```

logged response without gzip encoding:
```
{ statusCode: 201,
  body: 
   { ok: true,
     [...]
     token: '[...]' },
  headers: 
   { 'content-type': 'application/json',
     'cache-control': 'max-age=300',
     [...]
   },
   [...]
}
```

Solution was to use the parsed body, which is provided by npm-registry-client as a callback parameter.

PSanetra

Edit: This PR probably fixes https://github.com/semantic-release/cli/issues/190

gr2m commented:
hey :wave: thanks for sending the PR 👍 I think there is a linting error, could you please double check that? I’m vacationing until 22nd, could someone please set @PSanetra’s fork locally and run the setup to make sure if it works as expected? Our test coverage is ... not good right now :D

pvdlg commented:
I'll look at it tonight.
@PSanetra if you don't mind I'll cherry pick your commit and add some simplification around the existing code.
I will remove the complexity of using callback wrapped into `await new Promise(resolve => {...}`, and remove the `.then` and `.catch` and use a more async/await syntax with `try/catch` blocks.

PSanetra commented:
@gr2m I fixed the linting issue in the first commit.
@pvdlg I added another commit, which simplifies the code in my opinion, but you may also cherry pick the first commit and ignore the second commit.
